### PR TITLE
Cleanup KVS tests

### PR DESCRIPTION
**Pull request by chu11**
This PR cleans up a lot of the KVS tests, as legacy testing lead to duplicate code, duplicate tests, and unnecessary stuff.  As well as fixing up #1127, this should make it easier to convert the KVS tests
over to treeobj at a later date.

In the middle of all this, I added the ```-f``` option to ```flux kvs unlink```, as there are several circumstances where we want to delete stuff before a test, but don't want ```flux kvs unlink``` to return an error if that key/directory is missing.

Note that I'd like to rename the files ```t1000-kvs-basic.t``` and ```t1002-kvs-cmd.t```, as the tests are no longer "basic" and the "cmd" tests are no longer just about the kvs command.  I'll rename appropriately at a later time (probably at a patch to end of this PR), as it may be hard to review this PR if I rename the files and all my changes appears to be new files.

### CODE REVIEW

**Comment by chu11** [comment]
Huh, I got 3 build failures in travis. One hit #997, so I restarted and that passed.  But in the other ones, all unit tests passed.  So I'm not entirely sure why it failed.  Travis didn't appear to time out either.

**Comment by grondo** [comment]
Search for `write error` in the output? Could be #1145.

**Comment by chu11** [comment]
@grondo Thanks, seems to be it.  ```/bin/bash: line 5: echo: write error: Resource temporarily unavailable```

File: t/t1000-kvs-basic.t [hunk 1]
```diff
@@ -359,54 +359,54 @@ test_expect_success 'kvs: symlink: kvs_copy does not follow symlinks (bottom)' '
 '
 
 test_expect_success 'kvs: get_symlinkat works after symlink unlinked' '
-	${KVSBASIC} unlink $TEST &&
+	flux kvs unlink -Rf $TEST &&
 	flux kvs link $TEST.a.b.X $TEST.a.b.link &&
 	ROOTREF=$(${KVSBASIC} get-treeobj .) &&
-	${KVSBASIC} unlink $TEST &&
+	flux kvs unlink -R $TEST &&
```
Comment (garlick) [inline t/t1000-kvs-basic.t]: did a `-f` get missed here?

File: t/t1000-kvs-basic.t [hunk 2]
```diff
@@ -665,7 +665,7 @@ test_expect_success 'kvs: async kvs_fence allows puts with fence in progress' '
 
 # base64 data
 test_expect_success 'kvs: copy-tokvs and copy-fromkvs work' '
-	${KVSBASIC} unlink $TEST &&
+	flux kvs unlink -R $TEST &&
```
Comment (garlick) [inline t/t1000-kvs-basic.t]: missed `-f` here too?

**Comment by garlick** [comment]
Looks good - I probably should have removed the duplicate code/tests when I made the last round of KVS API changes.  Thanks for doing that.  I'm OK with renaming the tests too.

File: t/t1000-kvs-basic.t (hunk 2 above)
Comment (chu11) [inline t/t1000-kvs-basic.t]: yup thanks

File: t/t1000-kvs-basic.t (hunk 1 above)
Comment (chu11) [inline t/t1000-kvs-basic.t]: I think we keep this one at ```-R``` b/c we want the unlink to pass given the path created above it.

**Comment by chu11** [comment]
rebased on master and fixed the unlink -R @garlick mentions above.

renamed ```t1002-kvs-cmd.t``` to `t1000-kvs.t``` as it is now the main kvs test.  ```t1000-kvs-basic.t``` is now ```t1002-kvs-extra.t```, as it handles stress testing and other testing.

**Comment by grondo** [comment]
restarted one of the builds which hit the `write error` issue

**Comment by grondo** [comment]
Looks good and since @garlick already reviewed this I'll merge it unless I hear otherwise

**Pull request merged**